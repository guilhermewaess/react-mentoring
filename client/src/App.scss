html,
body,
.app {
  height: 100vh;
  overflow-y: auto;
}
