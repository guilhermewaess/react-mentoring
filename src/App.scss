html,
body,
.app {
  height: 100vh;
}
