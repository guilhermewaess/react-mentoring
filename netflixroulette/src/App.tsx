import * as React from 'react';
import { BrowserRouter as Router, Redirect, Route } from 'react-router-dom';
import { Container } from 'reactstrap';
import './App.scss';

import ErrorBoundary from './core/ErrorBoundary';
import MovieDetailsView from './views/MovieDetailsView';
import MoviesView from './views/MoviesView';

const RootRedirect = () => <Redirect to="/movies" />;

class App extends React.Component {
  public render() {
    return (
      <Container fluid={true} className="p-0 m-0 fixed-top">
        <ErrorBoundary>
          <Router>
            <div className="app">
              <Route exact={true} path="/" component={RootRedirect} />
              <Route path="/movies" component={MoviesView} />
              <Route path="/movie-details" component={MovieDetailsView} />
            </div>
          </Router>
        </ErrorBoundary>
      </Container>
    );
  }
}

export default App;
