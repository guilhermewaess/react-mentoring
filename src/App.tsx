import * as React from 'react';
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from 'react-router-dom';
import { Container } from 'reactstrap';
import './App.scss';

import ErrorBoundary from './core/ErrorBoundary';
import MovieDetailsView from './views/MovieDetailsView';
import MoviesView from './views/MoviesView';

class App extends React.Component {
  public render() {
    return (
      <Container fluid={true} className="p-0 m-0 fixed-top">
        <ErrorBoundary>
          <Router>
            <div className="app">
              <Switch>
                <Route path="/movies" component={MoviesView} />
                <Route path="/movie-details" component={MovieDetailsView} />
                <Redirect from='/' to='/movies' />
              </Switch>
            </div>
          </Router>
        </ErrorBoundary>
      </Container>
    );
  }
}

export default App;
