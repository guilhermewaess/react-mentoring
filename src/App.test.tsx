import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import * as configureStore from 'redux-mock-store';
import * as reduxPromise from 'redux-promise';
import App from './App';


describe('App', () => {
  let app: ShallowWrapper;
  let store: any;

  beforeEach(() => {
    store = configureStore([reduxPromise])({});
    app = shallow(<App store={store}/>);
  });
  it('should render', () => {
    expect(shallowToJson(app)).toMatchSnapshot();
  });
});
