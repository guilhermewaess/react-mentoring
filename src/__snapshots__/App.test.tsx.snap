// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Container
    className="p-0 m-0 fixed-top"
    fluid={true}
    tag="div"
  >
    <ErrorBoundary>
      <BrowserRouter>
        <div
          className="app"
        >
          <Switch>
            <Route
              component={[Function]}
              path="/movies"
            />
            <Route
              component={[Function]}
              path="/movie-details"
            />
            <Redirect
              from="/"
              push={false}
              to="/movies"
            />
          </Switch>
        </div>
      </BrowserRouter>
    </ErrorBoundary>
  </Container>
</Provider>
`;
